pymc3/distributions/bound.py

```diff
@@ -263,47 +75,61 @@ class Bound:
-                "Observed Bound distributions are not supported. "
-                "If you want to model truncated data "
-                "you can use a pm.Potential in combination "
-                "with the cumulative probability function."
-            )
+        # if "observed" in kwargs:
+        #     raise ValueError(
+        #         "Observed Bound distributions are not supported. "
+        #         "If you want to model truncated data "
+        #         "you can use a pm.Potential in combination "
+        #         "with the cumulative probability function."
+        #     )
 
-        transform = kwargs.pop("transform", "infer")
-        if issubclass(self.distribution, Continuous):
-            return _ContinuousBounded(
-                name, self.distribution, self.lower, self.upper, transform, *args, **kwargs
-            )
-        elif issubclass(self.distribution, Discrete):
-            return _DiscreteBounded(
-                name, self.distribution, self.lower, self.upper, transform, *args, **kwargs
-            )
+        transform = kwargs.pop("transform", None)
+        if issubclass(distribution, Continuous):
+            if transform == None:
+                if lower is not None and upper is not None:
+                    transform = transforms.interval(lower, upper)
+                    default = 0.5 * (lower + upper)
+                elif upper is not None:
+                    transform = transforms.upperbound(upper)
+                else:
+                    transform = transforms.lowerbound(lower)
+
+        elif issubclass(distribution, Discrete):
+            raise ValueError("Can not transform discrete variable.")
         else:
             raise ValueError("Distribution is neither continuous nor discrete.")
 
-    def dist(self, *args, **kwargs):
-        if issubclass(self.distribution, Continuous):
-            return _ContinuousBounded.dist(
-                self.distribution, self.lower, self.upper, *args, **kwargs
-            )
+        if transform is not None:
+            kwargs["transform"] = transform
+        return super.dist([distribution, lower, upper], **kwargs)
+
+    def logp(value, distribution, lower, upper):
+        """
+        Calculate log-probability of Bounded distribution at specified value.
+
+        Parameters
+        ----------
+        value: numeric
+            Value for which log-probability is calculated.
 
-        elif issubclass(self.distribution, Discrete):
-            return _DiscreteBounded.dist(self.distribution, self.lower, self.upper, *args, **kwargs)
+        Returns
+        -------
+        TensorVariable
+        """
+        logp = _logp(distribution.owner.op, {}, value, *distribution.owner.inputs[3:])
+        bounds = []
+        if lower is not None:
+            bounds.append(value >= lower)
+        if upper is not None:
+            bounds.append(value <= upper)
+        if len(bounds) > 0:
+            return bound(logp, *bounds)
```

Comment:
How is the `bounds` function different from `interval` ?

Reply:
You mean `bound` from `distributions.dist_math`? It simply wraps a switch around the `logp` expression, which returns `-inf` if any of the bounds is not respected. In this case it's basically `at.switch((value < lower) or (value > upper), -np.inf, logp)`